includes/tracks/events/class-wc-admin-setup-wizard-tracking.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * WooCommerce Admin Setup Wizard Tracking
+ *
+ * @package WooCommerce\Tracks
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class adds actions to track usage of the WooCommerce Onboarding Wizard.
+ */
+class WC_Admin_Setup_Wizard_Tracking {
+	/**
+	 * Init tracking.
+	 */
+	public static function init() {
+		if ( empty( $_GET['page'] ) || 'wc-setup' !== $_GET['page'] ) { // WPCS: CSRF ok, input var ok.
+			return;
+		}
+
+		add_action( 'admin_init', array( __CLASS__, 'track_store_setup' ), 1 );
+	}
+
+	/**
+	 * Check if a step is being saved by step name.
+	 *
+	 * @param string $step_name Step name.
+	 * @return bool
+	 */
+	public static function is_saving_step( $step_name ) {
+		// phpcs:disable WordPress.Security.NonceVerification.NoNonceVerification
+		$current_step = isset( $_GET['step'] ) ? sanitize_key( $_GET['step'] ) : '';
+		if ( ! empty( $_POST['save_step'] ) && $current_step === $step_name ) {
+			return true;
+		}
+		// phpcs:enable
+
+		return false;
+	}
+
+	/**
+	 * Track store setup and store properties on save.
+	 *
+	 * @return void
+	 */
+	public static function track_store_setup() {
+		// First step name is blank.
+		if ( ! self::is_saving_step( '' ) ) {
```

Comment:
When I go back to the first step after having already completed it, there is a step `&step=store_setup`.

Reply:
Regarding `*_obw_start`:

>User clicks the button to start the Onboarding Wizard

This would be the "activate" button once the plugin is downloaded.

However, on a fresh install, a user is sent to the OBW without the `step` query parameter. We can hook into the page view and check if the step parameter is missing to fire off the `*_obw_start` event. Would that work?

Reply:
Nice catch on that step var being set after the fact.  Added that to the switch statement.